cl/search/tasks.py

```diff
@@ -120,29 +124,35 @@ def add_docket_to_solr_by_rds(item_pks, force_commit=False):
     needed).
     :return: None
     """
-    si = scorched.SolrInterface(settings.SOLR_RECAP_URL, mode="w")
-    rds = RECAPDocument.objects.filter(pk__in=item_pks).order_by()
-    try:
-        metadata = rds[0].get_docket_metadata()
-    except IndexError:
-        metadata = None
-
-    try:
-        si.add([item.as_search_dict(docket_metadata=metadata) for item in rds])
-        if force_commit:
-            si.commit()
-        si.conn.http_connection.close()
-    except SolrError as exc:
-        add_docket_to_solr_by_rds.retry(exc=exc, countdown=30)
+    with Session() as session:
+        si = scorched.SolrInterface(
+            settings.SOLR_RECAP_URL, http_connection=session, mode="w"
+        )
+        rds = RECAPDocument.objects.filter(pk__in=item_pks).order_by()
+        try:
+            metadata = rds[0].get_docket_metadata()
+        except IndexError:
+            metadata = None
+
+        try:
+            si.add(
```

Comment:
Similar thought here. Worth a light refactor to put the context block around the next couple lines?

Reply:
I'm confused what you're suggesting here, I basically made sure everything using the `SolrInterface` was inside the associated session context manager block already.